src/main/java/com/example/pokebattlez/battle/controller/BattleController.java

```diff
@@ -17,17 +21,32 @@ public BattleController(BattleService battleService) {
     }
 
     @MessageMapping("/{battleId}/move")
-    public void chooseMove(@DestinationVariable String battleId, @RequestParam("id") Long playerId, @RequestParam("index") Integer index, @RequestParam("target") Long targetId) {
-        battleService.useMove(battleId, playerId, index, targetId);
+    public void chooseMove(@DestinationVariable String battleId, Request request) {
+        System.out.println(request);
+        battleService.useMove(battleId, request.playerId, request.move, request.targetId);
     }
 
     @MessageMapping("/{battleId}/switch")
-    public void choosePokemon(@DestinationVariable String battleId, @RequestParam("id") Long playerId, @RequestParam("index") Integer index) {
-        battleService.switchPokemon(battleId, playerId, index);
+    public void choosePokemon(@DestinationVariable String battleId, Request request) {
+        battleService.switchPokemon(battleId, request.playerId, request.index);
     }
 
     @MessageMapping("/{battleId}/cancel")
-    public void cancelAction(@DestinationVariable String battleId, @RequestParam("id") Long playerId) {
-        battleService.cancelAction(battleId, playerId);
+    public void cancelAction(@DestinationVariable String battleId, Request request) {
+        battleService.cancelAction(battleId, request.playerId);
+    }
+
+    @MessageMapping("/{battleId}/connect")
+    public void connect(@DestinationVariable String battleId, Request request) {
+        battleService.connectTrainer(battleId, request.getPlayerId());
+    }
+
+    @Data
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static class Request {
+        private Long playerId;
+        private Integer index;
+        private String move;
+        private Long targetId;
```

Comment:
Why this class is here? Controllers should not have any model declarations. This is looks like a breach of MVC